runtime/cudaq/platform/default/rest/helpers/oqc/OQCServerHelp.cpp

```diff
@@ -0,0 +1,298 @@
+  std::string extractJobId(ServerMessage &postResponse) override;
+
+  /// @brief Constructs the URL for retrieving a job based on the server's
+  /// response to a job submission.
+  std::string constructGetJobPath(ServerMessage &postResponse) override;
+
+  /// @brief Constructs the URL for retrieving a job based on a job ID.
+  std::string constructGetJobPath(std::string &jobId) override;
+
+  /// @brief Constructs the URL for retrieving the results of a job based on the
+  /// server's response to a job submission.
+  std::string constructGetResultsPath(ServerMessage &postResponse);
+
+  /// @brief Constructs the URL for retrieving the results of a job based on a
+  /// job ID.
+  std::string constructGetResultsPath(std::string &jobId);
+
+  /// @brief Retrieves the results of a job using the provided path.
+  ServerMessage getResults(std::string &resultsGetPath);
+
+  /// @brief Checks if a job is done based on the server's response to a job
+  /// retrieval request.
+  bool jobIsDone(ServerMessage &getJobResponse) override;
+
+  /// @brief Processes the server's response to a job retrieval request and
+  /// maps the results back to sample results.
+  cudaq::sample_result processResults(ServerMessage &postJobResponse) override;
+
+  std::string get_from_config(BackendConfig config, std::string key);
+  std::string get_from_config(BackendConfig config, std::string key,
+                              std::string default_return);
+};
+
+// Initialize the OQC server helper with a given backend configuration
+void OQCServerHelper::initialize(BackendConfig config) {
+  cudaq::info("Initializing OQC Backend.");
+  // Move the passed config into the member variable backendConfig
+  backendConfig = std::move(config);
+  // Set the necessary configuration variables for the OQC API
+  backendConfig["url"] = OQCServerHelper::get_from_config(config, "url");
+  backendConfig["version"] = "v0.3";
+  backendConfig["user_agent"] = "cudaq/0.3.0";
+  backendConfig["target"] = "Lucy";
+  backendConfig["qubits"] = 8;
+  // Retrieve the API key from the environment variables
+  backendConfig["email"] = OQCServerHelper::get_from_config(config, "email");
+  backendConfig["password"] =
+      OQCServerHelper::get_from_config(config, "password");
+  // Construct the API job path
+  backendConfig["job_path"] = "/tasks"; // backendConfig["url"] + "/tasks";
+}
+
+std::string OQCServerHelper::get_from_config(BackendConfig config,
+                                             std::string key,
+                                             std::string default_return) {
+  std::string output = OQCServerHelper::get_from_config(config, key);
+  if (output.empty()) {
+    return default_return;
+  }
+  return output;
+}
+
+std::string OQCServerHelper::get_from_config(BackendConfig config,
+                                             std::string key) {
+  auto iter = backendConfig.find(key);
+  if (iter != backendConfig.end()) {
+    return iter->second;
+  }
+  return "";
+}
```

Comment:
I'd recommend merging these two functions, e.g., 
```
std::string OQCServerHelper::get_from_config(BackendConfig config, const std::string& key, const std::string& default_return = "") {
 auto iter = backendConfig.find(key);
  if (iter != backendConfig.end()) {
    return iter->second;
  }
  return default_return; 
}
```

Reply:
It looks like this was incorporated, so I will mark this as resolved.